var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -673,23 +761,20 @@ def cmake_args(self):
             projects.append("flang")
         if "+lld" in spec:
             projects.append("lld")
-        if "+compiler-rt" in spec:
-            if self.spec.satisfies("@15.0.0:"):
-                runtimes.append("compiler-rt")
-            else:
-                projects.append("compiler-rt")
-        if "+libcxx" in spec:
-            if self.spec.satisfies("@15.0.0:"):
-                runtimes.extend(["libcxx", "libcxxabi"])
-            else:
-                projects.extend(["libcxx", "libcxxabi"])
+        if "compiler-rt=runtime" in spec:
```

Comment:
If we can standardize on this style, it might be nice to loop over keys in the spec with a given property, and add them to runtimes or projects based on their value.  Loving this refactor overall if we can make it fly. 👍

Reply:
Could you, please, give an example of how to "loop over keys in the spec with a given property"? I thought of refactoring it but the problem was that the variant `omp` adds `openmp` and the variant `libcxx` adds `libcxx` and `libcxxabi`. Currently we'd have something like:
```python
for var, params in {
    "omp": ["openmp"],
    "libunwind": ["libunwind"],
    "libcxx": ["libcxx", "libcxxabi"],
    "compiler-rt": ["compiler-rt"],
}.items():
    val = spec.variants[var].value
    if val == "project":
        projects.extend(params)
    elif val == "runtime":
        runtimes.extend(params)
```
Does it really make sense?